pkg/auth/user/helper/helper.go

```diff
@@ -97,3 +98,13 @@ func CreateErrorMessageForUserRoleGroups(restrictedGroups []bean2.RestrictedGrou
 func GetCasbinNameFromRoleGroupName(name string) string {
 	return "group:" + strings.ReplaceAll(strings.ToLower(name), " ", "_")
 }
+
+func CheckIfSuperAdminFromRoles(roles []*repository.RoleModel) bool {
+	isSuperAdmin := false
+	for _, role := range roles {
+		if role.Role == bean2.SUPERADMIN {
+			isSuperAdmin = true
```

Comment:
this will set the value from the last role

Reply:
it will only be set when role is superadmin and superadmin role will be only one

Reply:
but for optimisation will return